PyDynamic/model_estimation/fit_filter.py

```diff
@@ -87,92 +86,207 @@ def _fitIIR(
+    UHvals: Optional[np.ndarray] = None,
+    mc_runs: Optional[int] = 1000,
+) -> Union[
+    Tuple[np.ndarray, np.ndarray, int], Tuple[np.ndarray, np.ndarray, int, np.ndarray]
+]:
+    """Least-squares (time-discrete) IIR filter fit to a given frequency response or
+    its reciprocal
+
+    For the forward problem this method uses Gauss-Newton non-linear optimization.
+    For the inverse problem it uses the equation-error method. The filter then is
+    stabilized by pole mapping and introduction of a time delay. Associated
+    uncertainties are optionally propagated when provided using the Monte Carlo method.
+
     Parameters
     ----------
-        Hvals:   numpy array of (complex) frequency response values of shape (M,)
-        Nb:      integer numerator polynomial order
-        Na:      integer denominator polynomial order
-        f:       numpy array of frequencies at which Hvals is given of shape
-        (M,)
-        Fs:      sampling frequency
-        tau:     integer initial estimate of time delay
-        justFit: boolean, when true then no stabilization is carried out
-    
+    Hvals : array_like of shape (M,)
+        (Complex) frequency response values.
+    Nb : int
+        Order of IIR numerator polynomial.
+    Na : int
+        Order of IIR denominator polynomial.
+    f : array_like of shape (M,)
+        Frequencies at which `Hvals` is given.
+    Fs : float
+        Sampling frequency for digital IIR filter.
+    tau : int, optional
+        Initial estimate of time delay for filter stabilization (default = 0). If
+        `justFit = True` this parameter is not used and `tau = 0` will be returned.
+    justFit : bool, optional
+        If True then no stabilization is carried out, if False (default) filter is
+        stabilized. This parameter is only available for reasons of backward
+        compatibility. If you intend to skip stabilization please from now on use
+        `max_stab_iter = 0` and leave `justFit` untouched. `justFit` will be removed
+        in a future release.
+    verbose : bool, optional
+        If True (default) be more talkative on stdout. Otherwise no output is written
+        anywhere.
+    max_stab_iter : int, optional
+        Maximum count of iterations for stabilizing the resulting filter. If no
+        stabilization should be carried out, this parameter can be set to 0 (default =
+        50). As long as `justFit` still exists `max_stab_iter` parameter has no
+        effect if `justFit = True`.
+    inv : bool, optional
+        If False (default) apply the forward method, otherwise perform the fit for
+        the reciprocal of the frequency response.
+    UHvals : array_like of shape (2M, 2M), optional
+        Uncertainties associated with real and imaginary part of H.
+    mc_runs : int, optional
+        Number of Monte Carlo runs, which has to be >= 1 (default = 1000). Only
+        used if uncertainties `UHvals` are provided. Otherwise this input has no effect.
```

Comment:
```suggestion
        Number of Monte Carlo runs >= 1 (default = 1000). Only used if uncertainties 
         `UHvals` are provided. Otherwise this input has no effect.
```